packages/eslint-mdx/src/parser.ts

```diff
@@ -81,10 +85,39 @@ export class Parser {
     this.parseForESLint = this.parseForESLint.bind(this)
   }
 
-  normalizeJsxNode(node: Node, parent?: Parent, options = this._options) {
+  // overload type, node type will return same type, union will return same union
+  normalizeJsxNode(
+    node: Jsx,
+    parent?: Parent,
+    options?: ParserOptions,
+  ): Jsx | Jsx[]
+
+  normalizeJsxNode(
+    node: Import,
+    parent?: Parent,
+    options?: ParserOptions,
+  ): Import
+
+  normalizeJsxNode(
+    node: Export,
+    parent?: Parent,
+    options?: ParserOptions,
+  ): Export
+
+  normalizeJsxNode(
+    node: Export | Import | Jsx,
+    parent?: Parent,
+    options?: ParserOptions,
+  ): Jsx[] | (Export | Import | Jsx)
+
+  normalizeJsxNode(
+    node: Export | Import | Jsx,
+    parent?: Parent,
+    options: ParserOptions = this._options,
+  ): Jsx[] | (Export | Import | Jsx) {
```

Comment:
Alternatively this method could be updated to only accept `Jsx` nodes and the filtering could be done ahead of time.

Reply:
That would be nicer, the current overload one seems too verbose.